tools/e2e_test_job.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###
+# This script is used by the E2E test job defined in .build/e2e_test.yaml
+# to prepare the Cloud Build environment and run the end-to-end tests.
+#
+
+set -euxo
+
+E2E_PROJECT_ID=cloud-sql-operator-testing
+
+# Install GCC and other essential build tools
+apt-get update
+apt-get install -y zip unzip build-essential
```

Comment:
How can we cache these steps such that we're not re-installed all these deps (and the ones below) for each e2e test build?

Reply:
I had the same question so I measured it. 

- Installing tools: 2 minutes
- Reconciling cloud resources with Terraform (no changes required): 3 minutes
- Building and deploying the operator image: 12 minutes
- Running e2e tests: 5 minutes. 

Overall: 21 minutes

Building and deploying is too slow. Mainly two steps make it slow: 

- Downloading all the go libraries used by the operator and the operator's build process. And it happens twice: once in the build container `make generate` and once again during `docker buildx build.` 
- Running the container build using Docker-in-docker. This is a lot slower than a native go compiler. 

I think we need to optimize this, but in a different PR. For now, I'd like to call this working for now, but not yet fast enough.

Reply:
So basically every build of e2e is at least 21 minutes?

Reply:
For now, yes.